src/OpenTelemetry/Metrics/Builder/MeterProviderBuilderExtensions.cs

```diff
@@ -238,6 +238,9 @@ public static MeterProviderBuilder SetMaxMetricStreams(this MeterProviderBuilder
     /// <param name="meterProviderBuilder"><see cref="MeterProviderBuilder"/>.</param>
     /// <param name="maxMetricPointsPerMetricStream">Maximum number of metric points allowed per metric stream.</param>
     /// <returns>The supplied <see cref="MeterProviderBuilder"/> for chaining.</returns>
+#if EXPOSE_EXPERIMENTAL_FEATURES
+    [Obsolete("Use MetricStreamConfiguration.CardinalityLimit via the AddView API instead. This method will be removed in a future version.")]
```

Comment:
irrespective of if this API is removed in future, the SDK will still have a default 2000 limit for *all* metrics, except for those overridden explicitly via AddView, right?

Reply:
suppose to: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5328/files#:~:text=public%20const%20int%20DefaultCardinalityLimit%20%3D%202000%3B
We can add tests for that.

Reply:
I tweaked a test to cover this which actually exposed a bug! Here is the test & fix: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5328/commits/f1b62b0763253ae978b6e57ce999044d509810f4

Reply:
ah, thank you! :sweat_smile:

Reply:
> I tweaked a test to cover this which actually exposed a bug! Here is the test & fix: [f1b62b0](https://github.com/open-telemetry/opentelemetry-dotnet/commit/f1b62b0763253ae978b6e57ce999044d509810f4)

💯